src/System.Runtime.Extensions/tests/System/Convert.BoxedObjectCheck.cs

```
@@ -9,33 +10,69 @@ namespace System.Tests
+            yield return new object[] { Int16.MaxValue };
+            yield return new object[] { (Int16)0 };
+            yield return new object[] { UInt16.MinValue };
+            yield return new object[] { UInt16.MaxValue };
+            yield return new object[] { Int32.MinValue };
+            yield return new object[] { Int32.MaxValue };
+            yield return new object[] { (Int32)0 };
+            yield return new object[] { UInt32.MinValue };
+            yield return new object[] { UInt32.MaxValue };
+            yield return new object[] { Int64.MinValue };
+            yield return new object[] { Int64.MaxValue };
+            yield return new object[] { (Int64)0 };
+            yield return new object[] { UInt64.MinValue };
+            yield return new object[] { UInt64.MaxValue };
+            yield return new object[] { Char.MinValue };
+            yield return new object[] { Char.MaxValue };
+            yield return new object[] { (Char)0 };
+            yield return new object[] { Double.MinValue };
+            yield return new object[] { Double.MaxValue };
+            yield return new object[] { (Double)0 };
+            yield return new object[] { Single.MinValue };
+            yield return new object[] { Single.MaxValue };
+            yield return new object[] { (Single)0 };
+            yield return new object[] { Decimal.MinValue };
+            yield return new object[] { Decimal.MaxValue };
+            yield return new object[] { (Decimal)0 };
+            yield return new object[] { DateTime.MinValue };
+            yield return new object[] { DateTime.Now };
+            yield return new object[] { DateTime.MaxValue };
+        }
 
-            foreach (object obj in testValues)
+        [Theory]
+        [MemberData(nameof(DefaultToTypeValues))]
+        public static void TestConvertedCopies(object testValue)
+        {
+            Assert.All(DefaultToTypeValues(), input =>
```

Using Assert.All here and then silently catching exceptions for mis-matched types seems a bit odd to me. What's this method trying to test?